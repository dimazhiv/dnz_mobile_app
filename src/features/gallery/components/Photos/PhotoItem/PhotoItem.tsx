import React from 'react';
import { Image, StyleSheet, TouchableOpacity } from 'react-native';

export type PhotoItemProps = {
  uri: string;
  onPress: (id: string) => void;
};

function PhotoItem({ uri, onPress }: PhotoItemProps) {
  const handlePhotoPress = () => {
    onPress(uri);
  };

  return (
    <TouchableOpacity onPress={handlePhotoPress}>
      <Image source={{ uri: uri }} style={styles.image} />
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  image: {
    width: 300,
    height: 300
  }
});

export default PhotoItem;
