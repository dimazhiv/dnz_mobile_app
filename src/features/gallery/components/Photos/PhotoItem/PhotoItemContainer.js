import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PhotoItem from './PhotoItem';
import { onPhotoPress } from '../../../sagaActions';

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      onPress: onPhotoPress
    },
    dispatch
  );
}

export default connect(null, mapDispatchToProps)(PhotoItem);
