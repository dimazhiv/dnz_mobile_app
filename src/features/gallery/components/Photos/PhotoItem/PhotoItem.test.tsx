import React from 'react';
import { shallow } from 'enzyme';
import PhotoItem from './PhotoItem';

describe('Snapshot testing PhotoItem component', () => {
  test('correct component view', () => {
    const tree = shallow(
      <PhotoItem uri={'https://farm9.staticflickr.com/8187/8432423659_dd1b834ec5.jpg'} onPress={() => {}} />
    );
    expect(tree).toMatchSnapshot();
  });
});
