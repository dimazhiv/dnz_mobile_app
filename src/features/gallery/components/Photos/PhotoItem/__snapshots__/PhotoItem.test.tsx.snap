// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing PhotoItem component correct component view 1`] = `
<TouchableOpacity
  onPress={[Function]}
>
  <Image
    source={
      {
        "uri": "https://farm9.staticflickr.com/8187/8432423659_dd1b834ec5.jpg",
      }
    }
    style={
      {
        "height": 300,
        "width": 300,
      }
    }
  />
</TouchableOpacity>
`;
